### DISCUSSION TITLE
Datahub Error adding configuration: Warning: touch(): Utime failed: Permission denied

### DISCUSSION THREAD

**Discussion by shades3002** [pimcore/pimcore#16362] [Q&A / Support] [answered]
I am experiencing this permissions error using datahub. Write permissions were added for the user www-data, the version of pimcore that I am using is 11.0.5, here is part of my composer.json: "pimcore/data-hub": "^1.6", "pimcore/data-importer ": "^1.8",

Here the error: Error adding configuration:
Warning: touch(): Utime failed: Permission denied

**Comment by kingjia90** [comment]
Could it be that you need also to change the permission via `chmod`?

**Reply by mohammed-mahmoud-scandi** [reply]
in pimcore 11, www-data now writes data in /config dir
```bash
chmod 777  ./config
```

**Comment by symos** [comment]
So, is it "official" that we have to give write permissions to the web server user on the config folder? This is not mentioned in the docs.

I would be very wary of giving access to the web server on the folder where the security.yaml file resides (not even discussing 777 rights).

**Comment by APochmann** [comment]
I ran into the same error code when trying to modify "Appearance & Branding" settings via UI. Although the setting was done it raised additionally the PHP Exception: "Warning: touch(): Utime failed: Permission denied".

The reason for this is, that in `LocationAwareConfigRepository->writeYaml` (which correctly modified `var/config/admin_system_settings/admin_system_settings.yaml`) there is a call of `invalidateConfigCache()` which runs touch command on `config/services.yaml` if the file exists. I personally have no idea why that is done but it causes the described exception. To my understanding there is no data in `config/services.yaml` which should be modified by the application so why this write access?

In my case the setting of both yaml files and their folders were identical in terms of owner and flags (owner apache and read/write flags for owner and group). The difference was the SELinux setting which allowed write access for httpd daemon only to files in `var` folder, but not in `config` folder, which I think is the correct setting and I fully agree with @symos that opening write access there is not a good idea.

**Comment by symos** [comment]
> I personally have no idea why that is done but it causes the described exception. To my understanding there is no data in config/services.yaml which should be modified by the application so why this write access?

When touching a file that already exists, its "last modified" entry changes. So, based on the function's name (invalidateConfigCache) we can deduce that Pimcore caches configuration as long as it's not been modified. When it wants to invalidate cache, it touches the file so that its last modified date gets updated and the config cache is built again. Not sure when it does this, nor why it chooses the services.yaml file, but it's not a case of the application wanting to modify the file's contents.

In any case, it'd be nice to have some official response as to what the correct permissions should be.

**Reply by kingjia90** [reply]
https://github.com/pimcore/pimcore/pull/18544#discussion_r2194709547 this may explain the reason behind this

**Comment by kingjia90** [comment, marked as answer]
https://github.com/pimcore/pimcore/pull/18544 might help to avoid the error
